source/deployment/sso-saml-adfs.rst

```diff
@@ -0,0 +1,191 @@
+
+  This action will add the signature to SAML messages, making verification successful.
+
+Export Identity Provider Certificate
+-------------------------------------
+
+Next, we export the identity provider certificate, which will be later uploaded to Mattermost to finish SAML configuration.
+
+1. In ADFS management sidebar, go to **AD FS > Service > Certificates** and double click on the certificate under **Token-signing**. You may alternatively right-click the field, then click **View Certificate...*
+
+	.. image:: ../../source/images/adfs_19_export_idp_cert_start.PNG
+
+2. In the **Certificate** screen, go to the **Details** tab and click **Copy to File…**, then **OK**. This opens a **Certificate Export Wizard**.
+
+	.. image:: ../../source/images/adfs_20_export_idp_cert_copy.PNG
+
+3. In the **Certificate Export Wizard** screen, click **Next**. Then, select the option **Base-64 encoded X.509 (.CER)** and click **Next** again.
+
+	.. image:: ../../source/images/adfs_21_export_idp_cert_wizard.PNG
+
+4. In the **Certificate Export Wizard** screen, click **Browse** to specify the location you want the Identity Provider Certificate to be exported, and specify the file name.
+
+	.. image:: ../../source/images/adfs_21-2_export_idp_cert_wizard.PNG
+
+5. Click **Save**. In the **Certificate Export Wizard** screen, verify the file path is correct, and click **Next**.
+
+6. In the **Completing the Certificate Export Wizard**, click **Finish**, then **OK** to confirm the export was successful.
+
+	.. image:: ../../source/images/adfs_21-3_export_idp_cert_wizard.PNG
+
+You’re now about to finish configuring SAML for Mattermost!
+
+Configure SAML for Mattermost
+-----------------------------
+
+1. Start Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**.
+  - *SAML SSO URL*: **SAML 2.0/W-Federation URL** ADFS Endpoint you copied earlier.
+  - *Identity Provider Issuer URL*: ``Relying party trust identifier`` from ADFS you specified earlier.
+  - *Identity Provider Public Certificate*: ``X.509 Public Certificate`` you downloaded earlier.
+
+	.. image:: ../../source/images/adfs_22_mattermost_basics.PNG
+
+2. (Optional) Configure Mattermost to verify the signature. The *Service Provider Login URL` is the `SAML 2.0 SSO service URL* you specified in ADFS earlier.
+
+	.. image:: ../../source/images/adfs_23_mattermost_verification.PNG
+
+3. (Optional) Enable encryption by uploading the Service Provider Private Key and Service Provider Public Certificate you generated earlier.
+
+	.. image:: ../../source/images/adfs_24_mattermost_encryption.PNG
+
+4. Set attributes for the SAML Assertions, which will be used to update user information in Mattermost. Attributes for email and username are required and should match the values you entered in ADFS earlier. See :ref:`documentation on SAML configuration settings <saml-enterprise>` for more detail.
```

Comment:
The :ref: is a Sphinx-specific markup, and is not part of reStructuredText, so it's not going to show up  correctly on GitHub preview. It does work when built locally with Sphinx, and should work on the docs site.